source/MaterialXView/Glsl/GLUtil/GLUtilityContext.cpp

```diff
@@ -0,0 +1,361 @@
+
+#include <MaterialXView/Glsl/GLUtil/HardwarePlatform.h>
+
+#if defined(OSWin_)
+#include <windows.h> // For Windows calls
+
+#elif defined(OSLinux_)
+#include <X11/Intrinsic.h> // for XtDisplay etc
+
+#elif defined(OSMac_)
+#include <MaterialXView/Glsl/GLUtil/CocoaWrappers.h>
+#endif
+
+#include <MaterialXView/Glsl/external/GLew/glew.h>
+#include <MaterialXView/Glsl/GLUtil/GLUtilityContext.h>
+
+namespace MaterialX
+{
+
+// Global base context 
+GLUtilityContext* GLUtilityContext::_globalGLUtilityContext = nullptr;
+
+// Unsupported
+#if defined(OSUnsupported_)
+
+GLUtilityContext::GLUtilityContext(const WindowWrapper& /*windowWrapper*/, HardwareContextHandle /*sharedWithContext*/) :
+    _contextHandle(nullptr),
+    _isValid(false)
+{
+}
+
+#elif defined(OSWin_)
+//
+// Windows implementation
+//
+GLUtilityContext::GLUtilityContext(const WindowWrapper& /*windowWrapper*/, HardwareContextHandle sharedWithContext) :
+    _contextHandle(nullptr),
+    _isValid(false)
+{
+    // For windows, we need a HDC to create an OpenGL context.
+    // Create a dummy 1x1 window and use it' HDC.
+    _dummyWindow.create("__GL_BASE_CONTEXT_DUMMY_WINDOW__", 1, 1, nullptr);
+    WindowWrapper dummyWindowWrapper = _dummyWindow.windowWrapper();
+
+    if (dummyWindowWrapper.isValid())
+    {
+        // Use a generic pixel format to create the context
+        static PIXELFORMATDESCRIPTOR pfd =
```

Comment:
We don't care about the format of this as we're using offscreen buffers created elsewhere to render with